server/cmwell-ws/app/security/EagerAuthCache.scala

```diff
@@ -0,0 +1,125 @@
+/**
+* Copyright 2015 Thomson Reuters
+*
+* Licensed under the Apache License, Version 2.0 (the “License”); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+* an “AS IS” BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package security
+
+import javax.inject.{Inject, Singleton}
+
+import com.typesafe.scalalogging.LazyLogging
+import logic.CRUDServiceFS
+import play.api.libs.json.{JsValue, Json}
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.Try
+import cmwell.domain.{Everything, FileContent, FileInfoton, Infoton}
+import cmwell.fts.{PaginationParams, PathFilter}
+import cmwell.util.concurrent._
+
+@Singleton
+class EagerAuthCache @Inject()(crudServiceFS: CRUDServiceFS)(implicit ec: ExecutionContext) extends LazyLogging {
+  private[this] var oData: AuthData = AuthData.empty
+  private[this] var nData: AuthData = AuthData.empty
+
+  private def data(nbg: Boolean) = if(nbg) nData else oData
+
+  SimpleScheduler.scheduleAtFixedRate(0.seconds, 10.minutes)(load())
+
+  // TODO Do not Await.result... These should return Future[Option[JsValue]]]
+  def getRole(roleName: String, nbg: Boolean): Option[JsValue] = {
+    data(nbg).roles.get(roleName).orElse {
+      Await.result(directReadFallback(s"/meta/auth/roles/$roleName", nbg), 6.seconds).map { role =>
+          logger.info(s"AuthCache(nbg=$nbg) role $roleName was not in memory, but added to Map")
+          if(nbg)
+            nData = nData.copy(roles = nData.roles + (roleName -> role))
+          else
+            oData = oData.copy(roles = oData.roles + (roleName -> role))
+        role
+      }
+    }
+  }
+
+  def getUserInfoton(userName: String, nbg: Boolean): Option[JsValue] = {
+    data(nbg).roles.get(userName).orElse {
+      Await.result(directReadFallback(s"/meta/auth/users/$userName", nbg), 6.seconds).map { user =>
+        logger.info(s"AuthCache(nbg=$nbg) user $userName was not in memory, but added to Map")
+        if(nbg)
+          nData = nData.copy(users = nData.users + (userName -> user))
+        else
+          oData = oData.copy(users = oData.users + (userName -> user))
+        user
+      }
+    }
+  }
+
+  def invalidate(nbg: Boolean): Unit = load()
+
+  private def load(): Unit = {
+    retryUntil[AuthData](isSuccessful = !_.isEmpty, maxRetries = 30, delay = 10.seconds)(loadOnce(false)).map { data => oData = combiner(oData, data) }
```

Comment:
10 retries should be enough IMO.
also, if ES is unresponsive, FTS' future is timed out after 10 sec. you schedule 10 sec after failure the next invokation.
This results in (10 + 10 + delta) seconds per retry.
multiply by 30, you'll get over 10 minutes.